cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+	repo, err := publisher.GetRepo()
+	if err != nil {
+		return err
+	}
+
+	// TODO: if not published? New commit? Etc...
+	if err = publisher.Publish(repo, subnetName, vm.Alias, subnetYAML, vmYAML); err != nil {
+		return err
+	}
+
+	ux.Logger.PrintToUser("Successfully published")
+	return nil
+}
+
+// iterate the reposDir to check what repos already exist locally
+// if nothing is found, prompt the user for an alias for a new repo
+func getAlias(reposDir string) error {
+	// have any aliases already been defined?
+	if alias == "" {
+		matches, err := os.ReadDir(reposDir)
+		if err != nil {
+			return err
+		}
+		if len(matches) == 0 {
+			// no aliases yet; just ask for a new one
+			alias, err = getNewAlias()
+			if err != nil {
+				return err
+			}
+		} else {
+			// there are already aliases, ask how to proceed
+			options := []string{"Provide a new alias", "Pick from list"}
+			choice, err := app.Prompt.CaptureList("Don't know which repo to publish to. How would you like to proceed?", options)
+			if err != nil {
+				return err
+			}
+			if choice == options[0] {
+				// user chose to provide a new alias
+				alias, err = getNewAlias()
+				if err != nil {
+					return err
+				}
+				// double-check: actually this path exists...
+				if _, err := os.Stat(filepath.Join(reposDir, alias)); err == nil {
+					ux.Logger.PrintToUser("The repository with the given alias already exists locally. You may have already published this subnet there (the other explanation is that a different subnet has been published there).")
+					yes, err := app.Prompt.CaptureYesNo("Do you wish to continue?")
+					if err != nil {
+						return err
+					}
+					if !yes {
+						ux.Logger.PrintToUser("User canceled, nothing got published.")
+						return nil
+					}
+				}
+			} else {
+				aliases := make([]string, len(matches))
+				for i, a := range matches {
+					aliases[i] = a.Name()
+				}
+				alias, err = app.Prompt.CaptureList("Pick an alias", aliases)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ask for a new alias
+func getNewAlias() (string, error) {
+	return app.Prompt.CaptureString("Please provide an alias for the repository we are going to use")
+}
+
+func getRepoURL(reposDir string) error {
+	if repoURL == "" {
+		path := filepath.Join(reposDir, alias)
+		repo, err := git.PlainOpen(path)
+		if err != nil {
+			// TODO: probably a debug log; the alias might not have been created yet,
+			// so the repo might not exist, therefore just ignore it!
+		} else {
+			// there is a repo already for this alias, let's try to figure out the remote URL from there
+			conf, err := repo.Config()
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			remotes := make([]string, len(conf.Remotes))
+			i := 0
+			for _, r := range conf.Remotes {
+				// TODO: Usually there's only one URL for a remote, but more are supported - should we too?
+				remotes[i] = r.URLs[0]
```

Comment:
you can use append instead of an index, if using capacity in make. in that way you avoid introducing the index element as a extra posible cause of bugs (eg aliasing)

Reply:
Hehe, we have talked about this in the other PR. I still think the index one is the pattern I want to use.

Reply:
I'm good with indexing too